### PR TITLE
pagination feature works but loading post images now a bit buggy

### PR DESCRIPTION
**Pull request by craig-a-lee**
implemented pagination, added searching for workouts and searching for exercises within workouts

### CODE REVIEW

**Comment by craig-a-lee** [comment]
Demo: https://imgur.com/a/ISdtOgO

File: Wellthier/Cells/PostCell.h [hunk 1]
```diff
@@ -16,6 +16,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 @property (weak, nonatomic) IBOutlet UIButton *toProfileHiddenButton;
 @property (weak, nonatomic) IBOutlet UILabel *displayName;
+@property (weak, nonatomic) IBOutlet UILabel *username;
```
Comment (xueyike) [inline Wellthier/Cells/PostCell.h]: nit: for those UILabel items, you should name them with Label as postfix. For example, usernameLabel.
Same for UITexView, it should be postTextView.

File: Wellthier/Cells/PostCell.m [hunk 2]
```diff
@@ -20,7 +20,12 @@ - (void)setPostDetails: (Post *)post {
     if (post[@"image"]) {
         self.postImageView.file = post[@"image"];
         [self.postImageView loadInBackground];
+        self.postImageView.hidden = NO;
+    } else {
+        self.postImageView.image = nil;
```
Comment (xueyike) [inline Wellthier/Cells/PostCell.m:25]: You don't need this line.

Comment (craig-a-lee) [reply Wellthier/Cells/PostCell.m:25]: was getting glitchy behavior without the else statement where pictures from a separate post would be displayed in the cell of a post without a picture

**Pull request merged**